Zend/zend_language_parser.y

```diff
@@ -1075,7 +1102,13 @@ new_variable:
 ;
 
 member_name:
-		T_STRING		{ $$ = $1; }
+		identifier { $$ = $1; }
+	|	'{' expr '}'	{ $$ = $2; }
+	|	simple_variable	{ $$ = zend_ast_create(ZEND_AST_VAR, $1); }
+;
+
+property_name:
```

Comment:
Good point, I forgot that method names have a more limited set of allowed keywords :)